pkg/pod/script.go

```diff
@@ -45,12 +45,12 @@ var (
 	}
 )
 
-// convertScripts converts any steps that specify a Script field into a normal Container.
+// convertScripts converts any steps and sidecars that specify a Script field into a normal Container.
 //
 // It does this by prepending a container that writes specified Script bodies
 // to executable files in a shared volumeMount, then produces Containers that
 // simply run those executable files.
-func convertScripts(shellImage string, steps []v1alpha1.Step) (*corev1.Container, []corev1.Container) {
+func convertScripts(shellImage string, steps []v1alpha1.Step, sidecars []v1alpha1.Sidecar) (*corev1.Container, []corev1.Container, []corev1.Container) {
```

Comment:
Rather than amending the signature and adding the extra func to account for Sidecars, couldn't the caller (MakePod) simply execute this function twice?  Once for Steps and once for Sidecars?

Reply:
Yes you can definitely do the same thing that way.  I think in order to do that you need to change the signature of `convertScripts` anyways to include context about the namePrefix.  Would you prefer I changed it to that?

Reply:
That would be excellent, thank you @tomgeorge !